apps/web/src/scenes/UserGalleryPage/UserSharedCommunities.tsx

```diff
@@ -0,0 +1,136 @@
+import { useCallback, useMemo } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import InteractiveLink from '~/components/core/InteractiveLink/InteractiveLink';
+import { HStack } from '~/components/core/Spacer/Stack';
+import { BaseS } from '~/components/core/Text/Text';
+import { useTrack } from '~/contexts/analytics/AnalyticsContext';
+import { useModalActions } from '~/contexts/modal/ModalContext';
+import { UserSharedCommunitiesFragment$key } from '~/generated/UserSharedCommunitiesFragment.graphql';
+import { useIsMobileWindowWidth } from '~/hooks/useWindowSize';
+import { getUrlForCommunity } from '~/utils/getCommunityUrlForToken';
+
+import PaginatedCommunitiesList from '../Modals/PaginatedList/PaginatedCommunitiesList';
+
+type Props = {
+  queryRef: UserSharedCommunitiesFragment$key;
+};
+
+export const COMMUNITIES_PER_PAGE = 20;
+export default function UserSharedCommunities({ queryRef }: Props) {
+  const query = useFragment(
+    graphql`
+      fragment UserSharedCommunitiesFragment on GalleryUser {
+        __typename
+        sharedCommunities(first: $sharedCommunitiesFirst, after: $sharedCommunitiesAfter)
+          @connection(key: "UserSharedCommunitiesFragment_sharedCommunities") {
+          edges {
+            node {
+              __typename
+              ... on Community {
+                __typename
+                name
+                chain
+                contractAddress {
+                  address
+                }
+              }
+            }
+          }
+          pageInfo {
+            total
+          }
+        }
+        ...PaginatedCommunitiesListFragment
+      }
+    `,
+    queryRef
+  );
+
+  const sharedCommunities = useMemo(
+    () => query.sharedCommunities?.edges?.map((edge) => edge?.node) ?? [],
+    [query.sharedCommunities?.edges]
+  );
```

Comment:
same thing here with the removeNullValues

Reply:
👍 added removeNullValues to the sharedCommunities memo here too